src/components/home.js

```diff
@@ -1,39 +1,60 @@
 import React, { useState, useEffect } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { v4 as uuidv4 } from 'uuid';
-import { addBook, removeBook } from '../redux/books/bookSlice';
+import { fetchBooks } from '../redux/books/bookSlice';
 import InputBooks from './logic/bookform';
-import BookList from './logic/booklist';
 
 const Home = () => {
   const dispatch = useDispatch();
+  const bookList = useSelector((state) => state.book);
   function getInitialBooks() {
-    const temp = localStorage.getItem('books');
-    const savedBooks = JSON.parse(temp);
-    return savedBooks || [];
+    return bookList.bookItems;
   }
   const [books, setBooks] = useState(getInitialBooks());
   useEffect(() => {
-    const temp = JSON.stringify(books);
-    localStorage.setItem('books', temp);
+    dispatch(fetchBooks());
   }, [books]);
-  const deleteBook = (id) => {
-    setBooks([
-      ...books.filter((book) => book.id !== id, dispatch(removeBook())),
-    ]);
+  const deleteBook = async (id) => {
+    await fetch(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/kWYN1eenheRfyNQFSTzX/books/${id}`, {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    setBooks([...books, bookList.bookItems]);
   };
-  const addBookItem = (cat, tittle, author) => {
-    const newBook = dispatch(addBook({
-      cat,
-      tittle,
-      author,
-      id: uuidv4(),
-    }));
-    setBooks([...books, newBook.payload]);
+  const addBookItem = async (cat, tittle, author) => {
+    await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/kWYN1eenheRfyNQFSTzX/books', {
+      method: 'POST',
+      body: JSON.stringify({
+        category: cat,
+        title: tittle,
+        author,
+        item_id: uuidv4(),
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    setBooks([...books, bookList.bookItems]);
   };
```

Comment:
- [ ] We are required to simply refactor the add book function in the `bookSlice.js` file. This method should be implemented in the book slice file. And we should update the state in the store, not just the local state. We can use `useSelector` to fetch the updated list from the store.